pkg/cache/informers.go

```diff
@@ -158,25 +158,49 @@ func (c *Store) updatePod(oldObj interface{}, newObj interface{}) {
 }
 
 func (c *Store) deletePod(obj interface{}) {
-	pod := obj.(*v1.Pod)
-	_, ok := pod.Labels[modelIdentifier]
-	_, existed := c.metaPods.Load(pod.Name) // Make sure nothing left.
-	if !ok && !existed {
+	var namespace, name string
+	pod, ok := obj.(*v1.Pod)
+	if !ok {
+		tombstone, ok := obj.(cache.DeletedFinalStateUnknown)
```

Comment:
that's a good catch. I am considering whether it's better to use switch case for `cache.DeletedFinalStateUnknown` and `v1.Pod` which makes the code cleaner?

something like this. otherwise the v1.pod-> cache.DeletedFinalStateUnknown -> v1.pod chain is kind of messy.

```
switch obj := obj.(type) {
case *v1.Pod:
	namespace, name = obj.Namespace, obj.Name
	_, hasModelLabel = obj.Labels[modelIdentifier]

case cache.DeletedFinalStateUnknown:
	if pod, ok := obj.Obj.(*v1.Pod); ok {
		namespace, name = pod.Namespace, pod.Name
		_, hasModelLabel = pod.Labels[modelIdentifier]

```

Reply:
Great suggestion, updated